### ISSUE TITLE
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove

### ISSUE THREAD

**Issue by JulienPalard** [ansible/ansible#21982]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

ansible-playbook

##### ANSIBLE VERSION

```
ansible 2.3.0
  config file = /home/…/ansible.cfg
  configured module search path = Default w/o overrides

```

##### CONFIGURATION

forks = 100, pipelining = True

##### OS / ENVIRONMENT

```
Distributor ID:	Debian
Description:	Debian GNU/Linux 8.7 (jessie)
Release:	8.7
Codename:	jessie
```

##### SUMMARY

Sometimes, like once every 10 runs, I'm getting a whole lot of (like 20 times, all have the exact same address):

```
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7f51bb741950>
Traceback (most recent call last):
  File "/usr/lib/python3.5/weakref.py", line 117, in remove
TypeError: 'NoneType' object is not callable
```

##### STEPS TO REPRODUCE

I can reproduce it with a `--limit` on one host and a `--start-at-task` with an unexisting task, so the playbook is doing like nothing apart skipping hosts.

**Comment by bcoca** [comment]
without a way to reproduce this, it will be hard to figure out what this bug is

**Comment by abadger** [comment]
I believe that this is this upstream python3 bug: https://bugs.python.org/issue29519  It's only been fixed recently so it's not in any released versions of python3 yet.

Is this just python3 spewing warning information or is it causing playbooks to fail?

**Comment by JulienPalard** [comment]
Only python3 warnings, no playbooks fail, looks like it's the python3 issue, nice find @abadger, closing this.

**Comment by tkoeck** [comment]
Configuration

Debian Stretch 64 Bit
Ansible: Version: 2.6.3-1ppa~trust

I get the same error around every 5th command execution

tkoeck@tron-nb:~$ ansible all -a "uname -a"
tron.someurl.de | SUCCESS | rc=0 >>
Linux tron 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt9-3 (2015-04-23) x86_64 GNU/Linux

Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7f9bb732aa60>
Traceback (most recent call last):
  File "/usr/lib/python3.5/weakref.py", line 117, in remove
TypeError: 'NoneType' object is not callable
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7f9bb732aa60>
Traceback (most recent call last):
  File "/usr/lib/python3.5/weakref.py", line 117, in remove
TypeError: 'NoneType' object is not callable
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7f9bb732aa60

**Comment by JulienPalard** [comment]
@tkoeck It's a Python bug, not an ansible one, please try a Python >= 3.6.

**Comment by aknrdureegaesr** [comment]
The solution is a two-line change, see https://github.com/python/cpython/commit/6c961c5a5396ef125dbbc47c060272a5d12c1646 . According to https://bugs.python.org/issue29519, the fix seems to be present in Python 3.5.4, 3.6.1, and  2.7.14. I guess it's in 3.7 as well. I have successfully done that change to a virtualenv based on 3.5.3 (which has the bug).
